src/content/popup/popup.ts

```diff
@@ -492,3 +498,91 @@ function getPopupArrow(): HTMLElement | null {
     ? hostElem.shadowRoot.querySelector('.arrow')
     : null;
 }
+
+type HorizontalSwipeDirection = 'right' | 'left';
+
+function onHorizontalSwipe(
+  element: HTMLElement,
+  handler: (swipeDirection: HorizontalSwipeDirection) => void
+) {
+  // Min x distance traveled to be considered swipe
+  const xMinThreshold = 150;
+  // Max y distance that can be traveled before it is no longer considered a horizontal swipe
+  const yMaxThreshold = 100;
+  // Max time allowed to travel that distance
+  const allowedTime = 200;
+
+  let startTime = 0;
+  let startX: number;
+  let startY: number;
+
+  element.addEventListener(
+    'touchstart',
+    function (e) {
+      startX = e.changedTouches[0].pageX;
+      startY = e.changedTouches[0].pageY;
+      startTime = Date.now();
+    },
+    false
+  );
+
+  element.addEventListener(
+    'touchmove',
+    function (e) {
+      // Prevent dragging viewport
+      e.preventDefault();
+    },
+    false
+  );
+
+  element.addEventListener(
+    'touchend',
+    function (e) {
+      const touch = e.changedTouches[0];
+      const deltaX = touch.pageX - startX;
+      const deltaY = touch.pageY - startY;
+      const elapsedTime = Date.now() - startTime;
+
+      // Check that elapsed time is within specified, horizontal dist traveled >= threshold, and vertical dist traveled <= 100
```

Comment:
Nit: Mind wrapping to 80 characters?

Reply:
Yeah no prob. I can probably go ahead and add a lint rule for this too—are you cool with wrapping all comments to 80 chars?

Reply:
> Yeah no prob. I can probably go ahead and add a lint rule for this too—are you cool with wrapping all comments to 80 chars?

Thanks, that would be great! Yep, all comments to 80 characters (obviously URLs can go over that but I assume the lint rule would automatically account for that).

Reply:
I've added the `@stylistic/eslint-plugin-js` and a `max-len` rule for this. There is a similar rule built into ESLint but it's deprecated and this is the way the docs recommend doing it now.

It's not possible to set max line lengths for just comments, so I've gone ahead and set it to 80 for comments but 120 for code, plus some more liberties like not caring about long string literals.

Reply:
Wow, thank you! It just occurred to me that we have prettier set up to wrap to 80 characters so maybe what we need instead is just to run `prettier --check` as part of our CI? Would that work?

Reply:
Thanks! Yeah, I've just landed an update to the CI workflow to run prettier there now.